core/lib/merkle_tree/src/domain.rs

```diff
@@ -280,18 +295,31 @@ impl ZkSyncTree {
 
     fn extract_writes(
         logs: impl Iterator<Item = TreeLogEntry>,
-        kvs: impl Iterator<Item = (Key, ValueHash)>,
-    ) -> (Vec<InitialStorageWrite>, Vec<RepeatedStorageWrite>) {
+        kvs: impl Iterator<Item = (StorageKey, ValueHash)>,
+    ) -> (
+        Vec<InitialStorageWrite>,
+        Vec<RepeatedStorageWrite>,
+        Vec<StateDiffRecord>,
+    ) {
         let mut initial_writes = vec![];
         let mut repeated_writes = vec![];
+        let mut state_diffs = vec![];
         for (log_entry, (key, value)) in logs.zip(kvs) {
             match log_entry {
                 TreeLogEntry::Inserted { leaf_index } => {
                     initial_writes.push(InitialStorageWrite {
                         index: leaf_index,
-                        key,
+                        key: key.hashed_key_u256(),
                         value,
                     });
+                    state_diffs.push(StateDiffRecord {
+                        address: *key.address(),
+                        key: h256_to_u256(*key.key()),
+                        derived_key: StorageKey::raw_hashed_key(key.address(), key.key()),
```

Comment:
Just to understand: Is replacing these 3 fields with `StorageKey` incorrect from the domain perspective? AFAICT, a `StorageKey` would contain data sufficient to restore these fields.

Reply:
Not incorrect, the basis for this was from test harness. It definitely could be updated but I'm not sure there's enough benefit to doing it now